File: src/main/javabak/machine_abstractions/ItemCapableBlockEntity.java
package redcrafter07.processed.block.machine_abstractions;

import net.minecraft.world.level.block.state.BlockState;
import net.neoforged.neoforge.items.IItemHandler;

import javax.annotation.Nullable;

public interface ItemCapableBlockEntity {
    @Nullable
    IItemHandler itemCapabilityForSide(@Nullable BlockSide side, BlockState state);
}

File: src/main/kotlin/redcrafter07/processed/block/machine_abstractions/ItemCapableBlockEntity.kt
package redcrafter07.processed.block.machine_abstractions

import net.minecraft.world.level.block.state.BlockState
import net.neoforged.neoforge.items.IItemHandler

interface ItemCapableBlockEntity {
    fun itemCapabilityForSide(side: BlockSide?, state: BlockState): IItemHandler?
}